docs/architecture.rst

```diff
@@ -1,142 +1,243 @@
-    The management panel can be accessed in the `Den account page <https://www.run.house/account#organizations>`_.
-
-Resource Access Levels
-----------------------
-
-Runhouse lets you control the access levels you provide to the users of your resources:
-
-- :code:`read` (default): Users cannot modify the config, but can access the resource directly.
-- :code:`write`: Users have full control over the resource, including modifying its config.
-
-
-Resource Visibility
--------------------
+    import runhouse as rh
+
+    # [1] and [2]
+    gpu = rh.cluster(name="rh-a10x", instance_type="A10G:1", provider="aws").up_if_not()
+
+    # [3]
+    sd_env = rh.env(reqs=["torch", "transformers", "diffusers"], name="sd_generate")
+    remote_sd_generate = rh.function(sd_generate).to(gpu, worker=sd_env)
+
+    # [4]
+    imgs = remote_sd_generate("A hot dog made out of matcha.")
+    imgs[0].show()
+
+    # [5]
+    remote_sd_generate.save()
+    sd_upsampler = rh.function(name="/my_username/sd_upsampler")
+    high_res_imgs = sd_upsampler(imgs)
+
+    # [6]
+    gpu.teardown()
+
+.. note::
+
+
+1. Specify and/or Allocate Compute
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Runhouse can allocate compute to the application on the fly, either by
+utilizing an existing VM or Ray cluster, or allocating a new one using local cloud or K8s credentials. The
+``rh.cluster`` constructor is generally used to specify and interact with remote compute, including bringing it up
+if necessary (``cluster.up_if_not()``).
+
+2. Starting the Runhouse Server Daemon
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If not already running, the client will start the Runhouse API server daemon
+on the compute and form a secure network connection (either over SSH or HTTP/S). Dependencies can be specified to be
+installed before starting the daemon.
+
+#. The daemon can be thought of as a "Python object server", holding key-value pairs of names and Python
+   objects in memory, and exposing an HTTP API to call methods on those objects by name.
+#. The objects are held in a single default worker process by default but can be sent to other worker
+   processes, including on other nodes in the cluster, to achieve powerful parallelism out of the box.
+#. If I call GET http://myserver:32300/my_object/my_method, the daemon will look up the object named
+   "my_object", issue an instruction for its worker to call the method "my_method" on it, and
+   return the result.
+#. The HTTP server and workers can handle thousands of concurrent calls per second, and have similar latency
+   under simple conditions to Flask.
+#. New workers can be constructed with ``rh.env``, which specifies the details of the Python environment
+   (packages, environment variables) in which the process will be constructed. By default, workers live
+   in the same Python environment as the daemon but can also be started in a conda environment or a
+   separate node.
+
+3. Deploying Functions or Classes
```

Comment:
in the snippet above we included `rh.env` in [3] and not [2]. should we briefly explain the env setup here as well?